prow/plank/controller.go

```diff
@@ -121,22 +121,47 @@ func (c *Controller) canExecuteConcurrently(pj *prowapi.ProwJob) bool {
 		}
 		if running >= max {
 			c.log.WithFields(pjutil.ProwJobFields(pj)).Debugf("Not starting another job, already %d running.", running)
-			return false
+			return false, nil
 		}
 	}
 
 	if pj.Spec.MaxConcurrency == 0 {
 		c.pendingJobs[pj.Spec.Job]++
-		return true
+		return true, nil
 	}
 
 	numPending := c.pendingJobs[pj.Spec.Job]
 	if numPending >= pj.Spec.MaxConcurrency {
 		c.log.WithFields(pjutil.ProwJobFields(pj)).Debugf("Not starting another instance of %s, already %d running.", pj.Spec.Job, numPending)
-		return false
+		return false, nil
 	}
+
+	// Check if there are more ProwJobs with an earlier creation timestamp than us and add them to numPending
+	// to ensure ProwJobs are executed serially by creationTimestamp
+	pjs, err := c.kc.ListProwJobs(c.selector)
```

Comment:
I think we should plumb through the list of jobs we already have instead of re-listing. Not sure if it's possible but doing one reconciliation loop over multiple lists might get weird, and list calls are very expensive anyway

Reply:
Or actually why not just use `c.pendingJobs`?

Reply:
Yeah, I realized that after creating the PR as well. `c.pendingJobs` doesn't do it, because we need to look into the jobs, but `c.pjs` is perfect